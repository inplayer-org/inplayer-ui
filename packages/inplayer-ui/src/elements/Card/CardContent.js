// @flow
import React from 'react';
import styled from 'styled-components';

const ContentWrapper = styled.div`
  flex: 3;
  padding: 1.2rem 0;
  height: 100%;
  width: 100%;
`;

const CardContent = ({ children }: Props) => <ContentWrapper>{children}</ContentWrapper>;

CardContent.defaultProps = {
  component: 'div',
};

export default CardContent;
