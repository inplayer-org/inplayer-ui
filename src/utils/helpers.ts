import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
