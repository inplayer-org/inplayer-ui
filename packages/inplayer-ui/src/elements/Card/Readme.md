```jsx static
import React from 'react';
import Card from '@inplayer-org/inplayer-ui';
import Typography from '@inplayer-org/inplayer-ui';

const SimpleCard = props => (
  <Card title={'Card with title'} titleVariant="h1" style={{ width: 400 }}>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
  </Card>
);
```

### Demo

```jsx
<Card title={'Card with title'} titleVariant="h1" style={{ width: 400 }}>
  <Typography variant="p">Card content</Typography>
  <Typography variant="p">Card content</Typography>
  <Typography variant="p">Card content</Typography>
</Card>
<br />
<Card style={{ width: 400 }}>
  <Typography variant="p">Card content</Typography>
  <Typography variant="p">Card content</Typography>
  <Typography variant="p">Card content</Typography>
</Card>
```
