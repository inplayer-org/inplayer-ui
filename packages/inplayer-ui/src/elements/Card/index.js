// @flow
import React, { type Node } from 'react';
import styled from 'styled-components';
import colors from 'config/colors';
import Typography from 'elements/Typography';
import CardContent from './CardContent';

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: relative;
  border: 1px solid ${colors.gray};
  border-radius: 3px;
  background: ${colors.white};
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0 1.5rem;
  list-style: none;
`;

const CardTitle = styled(Typography)`
  flex: 1;
  width: 100%;
  margin: 0;
  padding: 0.5rem 0;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: inline-block;
`;

type Props = {
  title?: string,
  titleVariant?: string,
  className?: string,
  children: Node,
  style?: Object,
};

const Card = ({ title, titleVariant, className, children, style }: Props) => (
  <CardWrapper className={className} style={style}>
    {title && <CardTitle variant={titleVariant}>{title}</CardTitle>}
    <CardContent>{children}</CardContent>
  </CardWrapper>
);

Card.defaultProps = {
  title: '',
  titleVariant: 'h1',
  className: '',
  style: {},
};

export default Card;
