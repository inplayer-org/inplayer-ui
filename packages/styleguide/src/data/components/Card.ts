import { ComponentData } from "../types";

const usage = `
  import React from 'react';
  import Card from '@inplayer-org/inplayer-ui';
  import Typography from '@inplayer-org/inplayer-ui';

  const SimpleCard = props => (
    <Card title={'Card with title'} titleVariant="h1" style={{ width: 400 }}>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
    </Card>
  );
`;

const demoCode = `<div>
  <Card title={'Card with title'} titleVariant="h1" style={{ width: 400 }}>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
  </Card>
  <br />
  <Card style={{ width: 400 }}>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
    <Typography variant="p">Card content</Typography>
  </Card>
</div>`;

const Card: ComponentData = {
  path: 'src/elements/Card/index.js',
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'children', type: 'Node', default: 'Required' },
    { name: 'className', type: 'string' },
    { name: 'style', type: 'Object', default: '{ }'},
    { name: 'title', type: 'string' },
    { name: 'titleVariant', type: 'string', default: 'h1' },
  ],
}

export default Card;