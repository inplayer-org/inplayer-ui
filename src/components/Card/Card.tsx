import React, { ReactChild } from 'react';
import styled from 'styled-components';
import colors from 'theme/colors';
import Typography, { TypographyVariant } from '../Typography/Typography';
import CardContent from './CardContent';

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: relative;
  border: 1px solid ${colors.gray};
  border-radius: 3px;
  background: ${colors.white};
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0 1.5rem;
  list-style: none;
`;

const CardTitle = styled(Typography)`
  flex: 1;
  width: 100%;
  margin: 0;
  padding: 0.5rem 0;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: inline-block;
`;

interface Props {
  /**
   * Title of the Card
   */
  title?: string;
  /**
   * Type of the variant
   */
  titleVariant?: TypographyVariant;
  className?: string;
  children: ReactChild;
}

const Card = ({ title, titleVariant = 'h1', className = '', children }: Props) => (
  <CardWrapper className={className}>
    {title && <CardTitle variant={titleVariant}>{title}</CardTitle>}
    <CardContent>{children}</CardContent>
  </CardWrapper>
);

export default Card;
