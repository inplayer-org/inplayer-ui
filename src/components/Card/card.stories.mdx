import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Card from './'
import Typography from '../Typography'

<Meta title="Card" component={<Card/>} />

# Card

```jsx
import { Card } from '@inplayer-org/inplayer-ui';

const Page = () => (
  <Card />
)
```

### Props
<Props of={Card}/>

### Simple Card
<Preview withSource="open">
  <Story name="simple card">
    <div>
        <Card title={'Card with title'} titleVariant="h2" style={{ width: 400 }}>
            <Typography variant="p">Card content</Typography>
            <Typography variant="p">Card content</Typography>
            <Typography variant="p">Card content</Typography>
        </Card>
    </div>
  </Story>
</Preview>

### No title Card
<Preview withSource="open">
  <Story name="only body">
    <div>
        <Card>
            <Typography variant="p">Card content</Typography>
            <Typography variant="p">Card content</Typography>
            <Typography variant="p">Card content</Typography>
        </Card>
    </div>
  </Story>
</Preview>

