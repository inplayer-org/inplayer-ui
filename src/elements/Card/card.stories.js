import React from 'react';
import { Typography, Colors } from 'src/index';
import styled from 'styled-components';
import Card from './index';

export default {
  component: Card,
  title: 'Card',
};
const StyledContainer = styled.div`
  width: 60vw;
  height: 60vh;
  margin-left: 100px;
  margin-top: 100px;
  border: 1px solid ${Colors.darkGray};
  background-color: ${Colors.gray};
`;
const HeadlineTypography = styled(Typography)`
  margin-left: 40px;
  margin-bottom: 0;
  color: ${Colors.darkBlue};
`;
const DescriptionTypography = styled(Typography)`
  margin-left: 40px;
  margin-bottom: 40px;
`;

export const SimpleCard = props => (
  <StyledContainer>
    <HeadlineTypography variant="h4">Card without title</HeadlineTypography>
    <DescriptionTypography variant="p" description>
      Cards contain content and actions about a single subject.
    </DescriptionTypography>
    <Card style={{ width: 400, marginLeft: 100 }}>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
    </Card>
  </StyledContainer>
);
export const WithTitle = props => (
  <StyledContainer>
    <HeadlineTypography variant="h4">Card with title</HeadlineTypography>
    <DescriptionTypography variant="p" description>
      Cards contain content and actions about a single subject.
    </DescriptionTypography>
    <Card title="Title" titleVariant="h1" style={{ width: 400, marginLeft: 100, marginTop: 40 }}>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
      <Typography variant="p">Card content</Typography>
    </Card>
  </StyledContainer>
);
