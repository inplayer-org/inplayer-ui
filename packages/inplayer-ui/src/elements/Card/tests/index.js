import { generateSnapshots } from 'packages/inplayer-ui/src/elements/Card/tests/helpers';

import Card from '../index';

generateSnapshots(Card);
