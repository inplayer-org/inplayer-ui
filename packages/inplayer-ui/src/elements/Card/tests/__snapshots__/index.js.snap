// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
.c1 {
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  padding: 0;
  height: 100%;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border: 1px solid #d7dde5;
  border-radius: 3px;
  background: #fff;
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0 1.5rem;
  list-style: none;
}

<div
  className="c0"
  style={Object {}}
>
  
  <div
    className="c1"
  />
</div>
`;
